Sources/Controllers/TargetMenu/GPX/TrackMenu/OABaseTrackMenuHudViewController.mm

```diff
@@ -290,13 +279,16 @@ - (BOOL)showStatusBarWhenFullScreen
 
 - (void)doAdditionalLayout
 {
-    self.backButtonLeadingConstraint.constant = [self isLandscape] ? self.tableView.frame.size.width : [OAUtilities getLeftMargin] + 10.;
+    self.backButtonLeadingConstraint.constant = [self isLandscape]
+            ? [self getLandscapeViewWidth] - [OAUtilities getLeftMargin] + 20.
+            : [OAUtilities getLeftMargin] + 10.;
     self.backButtonContainerView.hidden = ![self isLandscape] && self.currentState == EOADraggableMenuStateFullScreen;
 }
 
 - (void)adjustMapViewPort
 {
     _mapViewController.mapView.viewportXScale = [self isLandscape] ? VIEWPORT_SHIFTED_SCALE : VIEWPORT_NON_SHIFTED_SCALE;
```

Comment:
Add a check not to set these values if they are already at that value. These variables should not be changed frequently as they result in a complete redraw of the map.